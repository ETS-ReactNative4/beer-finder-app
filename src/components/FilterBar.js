import React from 'react';
import { StyleSheet, ScrollView, View, Text } from 'react-native';
import { Button } from 'react-native-elements';
import PropTypes from 'prop-types';

import BASE_COLOR, { WHITE_COLOR, GREY_COLOR } from '../helpers';

export default FilterBar = (props) => {
  return (
    <View style={styles.container}>
      <ScrollView
        horizontal
        contentContainerStyle={styles.scrollViewContent}
        showsHorizontalScrollIndicator={false}
      >
        <View style={{ flexDirection: 'row' }}>{props.children}</View>
        <Button
          title='Clear'
          onPress={props.clearFilters}
          buttonStyle={{
            alignSelf: 'flex-start',
            justifyContent: 'flex-end',
            backgroundColor: GREY_COLOR,
          }}
        ></Button>
      </ScrollView>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    paddingHorizontal: 10,
    paddingVertical: 5,
    backgroundColor: WHITE_COLOR,
    borderBottomColor: BASE_COLOR,
    borderBottomWidth: 0.5,
  },
  scrollViewContent: {
    flex: 1,
    justifyContent: 'space-between',
    alignItems: 'center',
  },
});

FilterBar.propTypes = {
  children: PropTypes.oneOfType([PropTypes.element, PropTypes.array]),
  clearFilters: PropTypes.func,
};
