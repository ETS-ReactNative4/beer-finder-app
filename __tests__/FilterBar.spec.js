import React from 'react';
import { create } from 'react-test-renderer';
import { Button } from 'react-native-elements';

import FilterBar from '../src/components/FilterBar';

describe('FilterBar component', () => {
  test('Matches the snapshot', () => {
    const filterBar = create(<FilterBar />);
    expect(filterBar.toJSON()).toMatchSnapshot();
  });

  test('it renders filters and clear button as children', () => {
    const component = create(
      <FilterBar clearFilters={this.clearFilters}>
        <Button title='Filter number one' />
        <Button title='Filter number two' />
        <Button title='Filter number two' />
        <Button title='Filter number two' />
      </FilterBar>
    );

    const instance = component.root;
    const buttons = instance.findAllByType(Button);

    expect(buttons).toHaveLength(5);
  });
});
