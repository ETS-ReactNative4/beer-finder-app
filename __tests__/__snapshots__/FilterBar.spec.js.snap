// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterBar component Matches the snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderBottomColor": "#000",
      "borderBottomWidth": 0.5,
      "flexDirection": "row",
      "paddingHorizontal": 10,
      "paddingVertical": 5,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "space-between",
      }
    }
    horizontal={true}
    showsHorizontalScrollIndicator={false}
  >
    <View>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      />
      <View
        style={
          Object {
            "borderRadius": 3,
            "overflow": "hidden",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "busy": false,
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "flex-start",
                "backgroundColor": "#CED0CE",
                "borderColor": "#2089dc",
                "borderRadius": 3,
                "borderWidth": 0,
                "flexDirection": "row",
                "justifyContent": "flex-end",
                "padding": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 18,
                  "paddingBottom": 1,
                  "paddingTop": 2,
                  "textAlign": "center",
                }
              }
            >
              Clear
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
